src/cdi_CPEloss/Tabular_CP_Eloss.hh

```diff
@@ -0,0 +1,107 @@
+
+#ifndef __cdi_CPEloss_Tabular_CP_Eloss_hh__
+#define __cdi_CPEloss_Tabular_CP_Eloss_hh__
+
+#include "cdi/CPCommon.hh"
+#include "cdi/CPEloss.hh"
+#include "ds++/Assert.hh"
+#include "ds++/DracoMath.hh"
+#include "experimental/mdspan"
+#include "units/PhysicalConstants.hh"
+#include "units/UnitSystemType.hh"
+#include <cmath>
+#include <fstream>
+#include <string>
+
+namespace rtt_cdi_cpeloss {
+//===========================================================================//
+/*!                                                                              
+ * \class Tabular_CP_Eloss                                                      
+ *                                                                               
+ * \brief Derived rtt_cdi::CPEloss class for tabular eloss.                     
+ *        This class implements the interface found in cdi/CPEloss.hh for        
+ *        the case where CP energy loss data is in tabular form, stored 
+ *        in a file.                
+ *                                                                               
+ */
+//===========================================================================//
+
+class Tabular_CP_Eloss : public rtt_cdi::CPEloss {
+public:
+  typedef std::vector<double> sf_double;
+
+private:
+  const std::string filename;
+  std::ifstream file;
+
+  rtt_units::UnitSystem us;
+  rtt_units::PhysicalConstants pc_cgs;
+
+  uint32_t n_energy;          //!< Number of gridpoints in projectile energy
+  uint32_t n_density;         //!< Number of gridpoints in target density
+  uint32_t n_temperature;     //!< Number of gridpoints in target temperature
+  double d_log_energy;        //!< Log spacing of projectile energy gridpoints
+  double d_log_density;       //!< Log spacing of target density gridpoints
+  double d_log_temperature;   //!< Log spacing of target temperature gridpoints
+  double min_log_energy;      //!< Log of minimum projectile energy
+  double min_log_density;     //!< Log of minimum target density
+  double min_log_temperature; //!< Log of minimum target temperature
+  double min_energy;          //!< Minimum target energy
+  double max_energy;          //!< Maximum target energy
+  double min_density;         //!< Minimum target density
+  double max_density;         //!< Maximum target density
+  double min_temperature;     //!< Minimum target temperature
+  double max_temperature;     //!< Maximum target temperature
+  sf_double energies;         //!< Vector of energy gridpoints
+  sf_double densities;        //!< Vector of density gridpoints
+  sf_double temperatures;     //!< Vector of temperature gridpoints
+  // Note that after unit conversions, *_energy is really *_speed
+
+  // Storage for tabulated data
+  std::experimental::basic_mdspan<double, std::experimental::extents<std::experimental::dynamic_extent, std::experimental::dynamic_extent, std::experimental::dynamic_extent>, std::experimental::layout_left> stopping_data;
```

Comment:
@KineticTheory I'm unhappy with this but I'm not sure how to get rid of the `std::experimental`s without adding to the namespace for all source files that include this header.